### PR TITLE
Make dot interpolation support quoted strings

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?243540

code buddy @czue 
app manager @snopoke / @dannyroberts

### CODE REVIEW

File: corehq/apps/app_manager/xpath.py [hunk 1]
```diff
@@ -29,10 +31,40 @@
...
+            if string[i] in "\"'":
+                # We're entering a quoted string: just record the quote type
+                quote = string[i]
+                new += string[i]
+            else:
+                # Non-quote character, not inside a quoted string
+                if string[i] == ".":
+                    # Replace dot with replacement, unless this looks like a decimal number
+                    if (i == 0 or re.match(r'\D', string[i - 1])):
+                        if (i == len(string)  - 1 or re.match(r'\D', string[i + 1])):
```
Comment (dimagimon) [inline corehq/apps/app_manager/xpath.py]: E221 multiple spaces before operator

**Review by czue — APPROVED** [review]
looks good and thorough. the implementation looks like it could get a little unwieldy to maintain (especially if this gets any more complicated). the tests are great and will help a lot. just wondering if we might be able to do something simpler like first extract all the quoted bits using something [like this](http://stackoverflow.com/a/19675957/8207) and then only run the interpolation on the rest, then rejoin everything back together? Maybe that's not any better but feels like it could be a little easier to reason with if it wasn't too hard.

**Comment by czue** [comment]
will let you self-merge or take a crack at my comment if it feels worth pursuing.

**Comment by orangejenny** [comment]
Agree that this is a finicky piece of code, but didn't find regex solutions any better. They get messy when dealing with quotes within quotes and then when rejoining the string. Going to leave this as-is.

**Pull request merged**